### PR TITLE
Sync upstream release 1.4

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Syncs `upbound-crossplane@release-1.4` with `crossplane@release-1.4`.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

`make reviewable`

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: .github/workflows/promote.yml [hunk 1]
```diff
@@ -16,6 +16,11 @@ env:
   # a step 'if env.DOCKER_USR' != ""', so we copy these to succinctly test whether
   # credentials have been provided before trying to run steps that need them.
   DOCKER_USR: ${{ secrets.DOCKER_USR }}
+<<<<<<< HEAD
```
Comment (tnthornton) [inline .github/workflows/promote.yml:19]: Left overs from your merge

Comment (hasheddan) [reply .github/workflows/promote.yml:19]: argghhh thank you!

Comment (hasheddan) [reply .github/workflows/promote.yml]: Fixed - thanks @tnthornton !

File: .github/workflows/configurations.yml [hunk 2]
```diff
@@ -0,0 +1,146 @@
+name: Configurations
```
Comment (turkenh) [inline .github/workflows/configurations.yml:1]: There is no configuration that we're publishing from this repository, hence we indeed removed this file: https://github.com/upbound/crossplane/pull/13

Does it just pop up when you follow https://github.com/upbound/universal-crossplane/blob/main/CONTRIBUTING.md#update-crossplane-version-skip-if-not-necessary ?

**Review by turkenh — APPROVED** [review]

File: .github/workflows/configurations.yml (hunk 2 above)
Comment (hasheddan) [reply .github/workflows/configurations.yml:1]: @turkenh thanks for catching! It came up in the merge conflicts commit https://github.com/upbound/crossplane/pull/33/commits/58f32bb70ffe584f2228af29aa571d59df71dc2e so I likely should have opted to remove but did not. Might be worth us adding a list of intentionally removed files to track this type of thing, but I'll fix in this PR and #32 for now! :)

**Pull request merged**